File: src/RuletaRusa/RuletaRusa.java
package RuletaRusa;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class RuletaRusa {
	private List<Integer> ruleta;
	private List<Integer> huecosBalas;
	private Scanner scan = new Scanner(System.in);
	private List<Integer> jugadores;
	
	public RuletaRusa() {
		this.ruleta = IntStream.rangeClosed(0, 5).boxed().collect(Collectors.toList());
		this.huecosBalas = new ArrayList<Integer>();
		this.jugadores = IntStream.rangeClosed(0, 1).boxed().collect(Collectors.toList());
	}
	
	public void girar() {
		Collections.shuffle(ruleta);
	}
	
	public void annadirNBalas(int n) {
		if(n <= 0 || n >= ruleta.size()) throw new IllegalArgumentException("Has llenado el cargador, nadie puede ganar");
	
		List<Integer> copiaRuleta = new ArrayList<Integer>(ruleta);
		Collections.shuffle(copiaRuleta);
		this.huecosBalas = copiaRuleta.subList(0, n);
	}
	
	public Integer disparar() {
		return ruleta.remove(0);
	}
	
	public void jugar() {
		System.out.println("Diga cuantas balas desea annadir:");
		Integer numBalas = scan.nextInt();
		System.out.printf("Se annadiran %d balas\n", numBalas);
		annadirNBalas(numBalas);
		int i = 0;
		while(!huecosBalas.isEmpty()) {
			System.out.printf("\nDiparar jugador %d\n", i);
			Integer balaDisparada = disparar();
			if(huecosBalas.contains(balaDisparada)) {
				System.out.printf("BANG!!!, muere jugador %d\n", i);
				jugadores.remove(i);
				break;
			}
			System.out.println("Click! Tuvo suerte");
			i = (i+1)%jugadores.size();
		}
		System.out.printf("Gana el jugador %d", jugadores.get(0));
 	}
	
	public static void main(String[] args) {
		RuletaRusa game = new RuletaRusa();
		game.jugar();
	}
}

File: src/python/RuletaRusa/RuletaRusa.py
from random import shuffle

class RuletaRusa(object):
    
    def __init__(self) -> None:
        self.__TAM_RULETA:int = 6 
        self.__NUM_JUGADORES:int = 2
        self.__ruleta:list[int] = list(range(0, self.__TAM_RULETA))
        self.__huecosBalas:list[int] = []
        self.__jugadores:list[int] = list(range(0, self.__NUM_JUGADORES))
        
    def girar(self) -> None:
        shuffle(self.__ruleta)
        
    def annadirNBalas(self, numBalas:int) -> None:
        copiaRuleta:list[int] = self.__ruleta.copy()
        shuffle(copiaRuleta)
        self.__huecosBalas = copiaRuleta[0: numBalas]
        
    def disparar(self) -> int:
        return self.__ruleta.pop()
    
    def play(self) -> None:
        print("Diga cuantas balas dese annadir:")
        numBalas:int = int(input())
        print(f"Se annadiran {numBalas} balas")
        self.annadirNBalas(numBalas)
        i:int = 0
        while(len(self.__huecosBalas) == 0):
           print(f"\nDisparar jugador {i}") 
           balaDisparada: int = self.disparar()
           if(self.__huecosBalas.__contains__(balaDisparada)):
               print(f"BANG!!!, muere jugador {i}")
               self.__jugadores.remove(i)
               break
           print("Click! Tuvo suerte")
           i = (i+1)%len(self.__jugadores)
        print(f"Gana el jugador {self.__jugadores[0]}")
        
        
if __name__ == "__main__":
    game: RuletaRusa = RuletaRusa()
    game.play()